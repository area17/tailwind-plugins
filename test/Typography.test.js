const generatePluginCss = require('./generatePluginCss');
const cssMatcher = require('jest-matcher-css');
const { Typography } = require('../index');
const helvetica400 = {
  'helvetica-400-1': {
    'font-size': 10,
    'line-height': 1.2,
    'font-weight': 400
  },

  'helvetica-400-2': {
    'font-size': 11,
    'line-height': 1.2,
    'font-weight': 400
  },

  'helvetica-400-3': {
    'font-size': 12,
    'line-height': 1.2,
    'font-weight': 400
  }
};

expect.extend({
  toMatchCss: cssMatcher
});

describe('typography plugin', () => {
  it('exists', () => {
    expect(typeof Typography).toBe('function');
  });

  test('generates correct css', () => {
    return generatePluginCss(Typography, {
      theme: {
        screens: {
          xs: { max: '543px' },
          sm: '544px',
          md: '768px',
          lg: '1024px',
          xl: '1440px'
        },

        typography: {
          family: {
            sans: 'Helvetica, Arial, sans-serif'
          },
          sets: {
            'body-1': {
              'font-family': 'sans',
              settings: {
                xs: helvetica400['helvetica-400-1'],
                md: helvetica400['helvetica-400-2'],
                lg: helvetica400['helvetica-400-3']
              }
            }
          }
        }
      }
    }).then((css) => {
      expect(css).toMatchCss(`
        .f-body-1 {
          font-family: Helvetica, Arial, sans-serif;
          font-size: 10px;
          line-height: 1.2;
          font-weight: 400
        }

        @media (min-width: 768px) {
          .f-body-1 {
            font-size: 11px;
            line-height: 1.2;
            font-weight: 400
          }
        }

        @media (min-width: 1024px) {
          .f-body-1 {
            font-size: 12px;
            line-height: 1.2;
            font-weight: 400
          }
        }
      `);
    });
  });

  test("doesn't output anything when no options set", () => {
    return generatePluginCss(Typography, {}).then((css) => {
      expect(css).toMatchCss(``);
    });
  });
});
