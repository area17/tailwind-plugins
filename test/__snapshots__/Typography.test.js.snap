// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`typography plugin generates correct css 1`] = `
":root {
    --sans: Helvetica, Arial, sans-serif;
    --f-h1-font-family: var(--sans);
    --f-h1-font-size: 2rem;
    --f-h1-font-stretch: initial;
    --f-h1-font-style: initial;
    --f-h1-font-variant: initial;
    --f-h1-font-weight: initial;
    --f-h1-line-height: 1.2;
    --f-h1-letter-spacing: -0.02em;
    --f-h1-font-feature-settings: initial;
    --f-h1--moz-osx-font-smoothing: grayscale;
    --f-h1--webkit-font-smoothing: antialiased;
    --f-h1---bold-weight: 500
}
.f-h1 {
    font-family: var(--f-h1-font-family);
    font-size: var(--f-h1-font-size);
    font-stretch: var(--f-h1-font-stretch);
    font-style: var(--f-h1-font-style);
    font-variant: var(--f-h1-font-variant);
    font-weight: var(--f-h1-font-weight);
    line-height: var(--f-h1-line-height);
    letter-spacing: var(--f-h1-letter-spacing);
    font-feature-settings: var(--f-h1-font-feature-settings);
    -moz-osx-font-smoothing: var(--f-h1--moz-osx-font-smoothing);
    -webkit-font-smoothing: var(--f-h1--webkit-font-smoothing);
    --bold-weight: var(--f-h1---bold-weight)
}
@media (min-width: 1024px) {
    :root {
        --f-h1-font-size: 3rem
    }
}
.f-h1 b, .f-h1 strong {
    font-weight: var(--f-h1---bold-weight, bold)
}
@media (min-width: 1024px) {
    .lg\\:f-h1 {
        font-family: var(--f-h1-font-family);
        font-size: var(--f-h1-font-size);
        font-stretch: var(--f-h1-font-stretch);
        font-style: var(--f-h1-font-style);
        font-variant: var(--f-h1-font-variant);
        font-weight: var(--f-h1-font-weight);
        line-height: var(--f-h1-line-height);
        letter-spacing: var(--f-h1-letter-spacing);
        font-feature-settings: var(--f-h1-font-feature-settings);
        -moz-osx-font-smoothing: var(--f-h1--moz-osx-font-smoothing);
        -webkit-font-smoothing: var(--f-h1--webkit-font-smoothing);
        --bold-weight: var(--f-h1---bold-weight)
    }
    .lg\\:f-h1 b,.lg\\:f-h1 strong {
        font-weight: var(--f-h1---bold-weight, bold)
    }
}"
`;
